### PR TITLE
rpc: paginate mempool /unconfirmed_txs endpoint

### LINKED ISSUE #7500: mempool: paginate /unconfirmed_txs rpc endpoint
## Summary

Unlimited maximum of http://localhost:26657/unconfirmed_txs, or the ability to paginate through them.

## Problem Definition

This feature would be useful for all of those trying to use and interact with the mempool of a slower chain with more pending transactions than the current maximum value of 100. This feature would still have the "limit" parameter so users could get as many or as little as they want without this limitation. I can't think of any disadvantages of removing this maximum, as there is a user set limit parameter which the user can choose.

## Proposal

There are two ways this problem can be solved, either adding a page feature so you can go paginate through the mempool, or removing this limit altogether
____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

### PR DESCRIPTION
**Pull request by spaceCh1mp**
This commit changes the behavior of the /unconfirmed_txs endpoint by replacing `limit` with a `page` and `perPage` parameter for pagination.
The test case for unconfirmed_txs has been modified to properly test this change and the documentation for the API as well.

fixes #7500
Closes #7500

### CODE REVIEW

File: internal/rpc/core/mempool.go [hunk 1]
```diff
@@ -117,19 +118,30 @@ func (env *Environment) BroadcastTxCommit(ctx context.Context, tx types.Tx) (*co
...
+// UnconfirmedTxs gets unconfirmed transactions
 // including their number.
 // More: https://docs.tendermint.com/master/rpc/#/Info/unconfirmed_txs
-func (env *Environment) UnconfirmedTxs(ctx context.Context, limitPtr *int) (*coretypes.ResultUnconfirmedTxs, error) {
+func (env *Environment) UnconfirmedTxs(ctx context.Context, pagePtr, perPagePtr *int) (*coretypes.ResultUnconfirmedTxs, error) {
 	// reuse per_page validator
-	limit := env.validatePerPage(limitPtr)
 
-	txs := env.Mempool.ReapMaxTxs(limit)
+	txs := env.Mempool.ReapMaxTxs(-1)
```
Comment (cmwaters) [inline internal/rpc/core/mempool.go:127]: This is a minor optimisation but say there are 5000 txs in the mempool and we've requested the first page of 100. This function will iterate over the entire mempool unnecessarily. Better would be to do something like:
```
totalCount := env.Mempool.Size()

. . .

txs := env.Mempool.ReapMaxTxs(skipCount+tmmath.MinInt(perPage, totalCount-skipCount)
result := txs[skipCount : ]
```

File: internal/rpc/core/mempool.go [hunk 2]
```diff
@@ -117,19 +118,30 @@ func (env *Environment) BroadcastTxCommit(ctx context.Context, tx types.Tx) (*co
 	}
 }
 
-// UnconfirmedTxs gets unconfirmed transactions (maximum ?limit entries)
+// UnconfirmedTxs gets unconfirmed transactions
 // including their number.
```
Comment (cmwaters) [inline internal/rpc/core/mempool.go:122]:
```suggestion
// UnconfirmedTxs gets unconfirmed transactions from the mempool in order of priority
```

**Review by cmwaters — COMMENTED** [review]
Nice work! I just have a few remarks.

File: internal/rpc/core/mempool.go (hunk 1 above)
Comment (spaceCh1mp) [reply internal/rpc/core/mempool.go:127]: This actually makes a lot of sense

**Review by cmwaters — APPROVED** [review]
Much appreciated 👏

**Pull request merged**